examples/sph-verlet/sph-main-verlet.cpp

```diff
@@ -0,0 +1,436 @@
+            sphSystem.updateHaloParticle(p);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * deletes the halo particles
+ * @param sphSystem
+ */
+void deleteHaloParticles(Container& sphSystem) {
+  sphSystem.deleteHaloParticles();
+}
+
+void densityPressureHydroForce(Container& sphSystem) {
+  // declare the used functors
+  autopas::sph::SPHCalcDensityFunctor densityFunctor;
+  autopas::sph::SPHCalcHydroForceFunctor hydroForceFunctor;
+
+  std::cout << "\nhaloupdate\n" << std::endl;
+
+  // 1.first calculate density
+  // 1.1 to calculate the density we need the halo particles
+  updateHaloParticles(sphSystem, sphSystem.needsRebuild());
+
+  // 1.2 then calculate density
+  for (auto part = sphSystem.begin(); part.isValid(); ++part) {
+    part->setDensity(0.);
+    densityFunctor.AoSFunctor(*part, *part);
+    part->setDensity(part->getDensity() / 2);
+  }
+  std::cout << "calculation of density... started" << std::endl;
+  sphSystem.iteratePairwiseAoS2(&densityFunctor);
+  std::cout << "calculation of density... completed" << std::endl;
+  // 1.3 delete halo particles, as their values are no longer valid
+  // deleteHaloParticles(sphSystem);
+
+  // 2. then update pressure
+  std::cout << "calculation of pressure... started" << std::endl;
+  setPressure(sphSystem);
+  std::cout << "calculation of pressure... completed" << std::endl;
+
+  // 0.3 then calculate hydro force
+  // 0.3.1 to calculate the density we need the halo particles
+  updateHaloParticles(sphSystem, false);
+
+  // 0.3.2 then calculate hydro force
+  for (auto part = sphSystem.begin(); part.isValid(); ++part) {
+    // self interaction leeds to:
+    // 1) vsigmax = 2*part->getSoundSpeed()
+    // 2) no change in acceleration
+    part->setVSigMax(2 * part->getSoundSpeed());
+    part->setAcceleration(std::array<double, 3>{0., 0., 0.});
+    part->setEngDot(0.);
+  }
+  std::cout << "calculation of hydroforces... started" << std::endl;
+  sphSystem.iteratePairwiseAoS2(&hydroForceFunctor);
+  std::cout << "calculation of hydroforces... completed" << std::endl;
+  // 0.3.3 delete halo particles, as their values are no longer valid
+  // deleteHaloParticles(sphSystem);
+}
+
+void printConservativeVariables(Container& sphSystem) {
+  std::array<double, 3> momSum = {0., 0., 0.};  // total momentum
+  double energySum = 0.0;                       // total energy
+  for (auto it = sphSystem.begin(autopas::IteratorBehavior::ownedOnly);
+       it.isValid(); ++it) {
+    momSum = autopas::arrayMath::add(
+        momSum, autopas::arrayMath::mulScalar(it->getV(), it->getMass()));
+    energySum += (it->getEnergy() +
+                  0.5 * autopas::arrayMath::dot(it->getV(), it->getV())) *
+                 it->getMass();
+  }
+  printf("%.16e\n", energySum);
+  printf("%.16e\n", momSum[0]);
+  printf("%.16e\n", momSum[1]);
+  printf("%.16e\n", momSum[2]);
+}
+
+int main() {
+  autopas::logger::create();
+
+  unsigned int rebuildFrequency = 6;
```

Comment:
would be cool if the simulation parameters could be set via input parameters but I guess that's not too relevant now ;)

Reply:
yup, later ^^